src/Compilers/CSharp/Portable/Binder/Binder.QueryUnboundLambdaState.cs

```diff
@@ -50,7 +50,7 @@ public override bool HasExplicitReturnType(out RefKind refKind, out TypeWithAnno
             public override RefKind RefKind(int index) { return Microsoft.CodeAnalysis.RefKind.None; }
             public override ScopedKind DeclaredScope(int index) => ScopedKind.None;
             public override MessageID MessageID { get { return MessageID.IDS_FeatureQueryExpression; } } // TODO: what is the correct ID here?
-            public override Location ParameterLocation(int index) { return _parameters[index].GetFirstLocation(); }
+            public override Location ParameterLocation(int index) { return _parameters[index].TryGetFirstLocation(); }
```

Comment:
I wasn't sure why we changed from `GetFirstLocation()` to `TryGetFirstLocation()` here. Maybe because we don't mind returning a possible null location here?

Reply:
correct.  RangeVariables actually support a null location.  This worked in teh past because it woudl actually wrap that null up in an ImmutableArray with one element in it.  This doesn't work in the new form as GetFirstLocation legit works by *trying* to get the location and throwing it is has none.  Since null and none are indistinguishable, this needed updating.   But it's a lot clearer IMO, and much better than having an array of one null location in it (which, imo, seems like it was a mistake in the first palce).